bpf/lib/process.h

```diff
@@ -409,6 +456,77 @@ static inline __attribute__((always_inline)) void execve_map_delete(__u32 pid)
 	}
 }
 
+// execve_joined_info_map_error() will increment the map error counter
+static inline __attribute__((always_inline)) void execve_joined_info_map_error(void)
+{
+	int one = MAP_STATS_ERROR;
+	__s64 *cntr;
+
+	cntr = map_lookup_elem(&tg_execve_joined_info_map_stats, &one);
+	if (cntr)
+		*cntr = *cntr + 1;
+}
+
+static inline __attribute__((always_inline)) int
+execve_joined_info_map_set(__u64 tid, struct execve_info *info)
+{
+	int err, zero = MAP_STATS_COUNT;
+	__s64 *cntr;
+
+	err = map_update_elem(&tg_execve_joined_info_map, &tid, info, BPF_ANY);
+	if (err < 0) {
+		/* -EBUSY or -ENOMEM with the help of the cntr error
+		 * on the stats map this can be a good indication of
+		 * long running workloads and if we have to make the
+		 * map size bigger for such cases.
+		 */
+		execve_joined_info_map_error();
+		return err;
+	}
+
+	cntr = map_lookup_elem(&tg_execve_joined_info_map_stats, &zero);
+	if (cntr && *cntr < EXECVE_JOINED_MAP_SIZE)
+		*cntr = *cntr + 1;
+
+	return err;
+}
+
+/* Clear up some space for next threads */
+static inline __attribute__((always_inline)) void
+execve_joined_info_map_clear(__u64 tid)
+{
+	int err, zero = MAP_STATS_COUNT;
+	__s64 *cntr;
+
+	err = map_delete_elem(&tg_execve_joined_info_map, &tid);
+	if (!err) {
+		cntr = map_lookup_elem(&tg_execve_joined_info_map_stats, &zero);
+		if (cntr)
+			*cntr = *cntr - 1;
+	}
+	/* We don't care here about -ENOENT as there is no guarantee entries
+	 * will be present anyway.
+	 */
+}
+
+static inline __attribute__((always_inline)) bool
+execve_joined_info_map_get(__u64 tid, struct execve_info *info, bool delete)
```

Comment:
Is the bool argument really needed here? There is only one caller.

Reply:
yes so reader knows get() could also delete

Reply:
Please use a different function name then. If the argument has always the same value, there is no need for it. `execve_joined_info_map_get_clear` for example.